src/protocol/vm/state.rs

```diff
@@ -664,6 +642,16 @@ impl VMPoolState<PreCachedDB> {
         }
 
         let buy_amount = trade.received_amount;
+
+        if sell_amount_exceeds_limit {
+            return Err(TychoSimulationError::SellAmountTooHigh(
+                // Partial buy amount and gas used
+                buy_amount,
+                trade.gas_used,
+                new_state,
+                sell_amount_limit,
+            ));
+        }
         Ok((buy_amount, trade.gas_used, new_state))
```

Comment:
Forgot to actually translate this to a `GetAmountOutResult`. Will do it tomorrow.